node/src/tasks/bsp_submit_proof.rs

```diff
@@ -268,21 +310,23 @@ where
                 RetryStrategy::default()
                     .with_max_retries(MAX_PROOF_SUBMISSION_ATTEMPTS)
                     .with_max_tip(max_tip as f64)
-                    .with_timeout(Duration::from_secs(self.storage_hub_handler.provider_config.extrinsic_retry_timeout))
+                    .with_timeout(Duration::from_secs(
+                        self.storage_hub_handler
+                            .provider_config
+                            .extrinsic_retry_timeout,
+                    ))
                     .with_should_retry(Some(Box::new(should_retry))),
-                false
+                false,
             )
             .await
             .map_err(|e| {
-                error!(target: LOG_TARGET, "CRITICAL❗️❗️ Failed to submit proof after {} attempts: {}", MAX_PROOF_SUBMISSION_ATTEMPTS, e);
-                anyhow!(
-                    "Failed to submit proof after {} attempts",
-                    MAX_PROOF_SUBMISSION_ATTEMPTS
-                )
+                error!(target: LOG_TARGET, "❌ Failed to submit proof due to: {}", e);
+                anyhow!("Failed to submit proof due to: {}", e)
             })?;
 
         trace!(target: LOG_TARGET, "Proof submitted successfully");
 
+        // TODO: Don't do this in this task any more.
```

Comment:
Where would you want to do it?

Reply:
It'll be done in every block, in the block import. We will run a `catch_up_to_forest_root_changes` (or some name like that).